lntest/itest/lnd_misc_test.go

```diff
@@ -1673,3 +1673,99 @@ func testSweepAllCoins(net *lntest.NetworkHarness, t *harnessTest) {
 		t.Fatalf("sweep attempt should fail")
 	}
 }
+
+// testListAddresses tests that we get all the addresses and their
+// corresponding balance correctly.
+func testListAddresses(net *lntest.NetworkHarness, t *harnessTest) {
+	ctxb := context.Background()
+	// First, we'll make a new node - Alice, which will be generating
+	// new addresses.
+	alice := net.NewNode(t.t, "Alice", nil)
+	defer shutdownAndAssert(net, t, alice)
+
+	// Next, we'll give Alice exactly 4 utxos of 1 BTC
```

Comment:
nit: missing full stop. Also, I think the comment isn't up to date anymore as we're only sending one UTXO.